airflow/models/connection.py

```diff
@@ -145,6 +145,38 @@ def parse_from_uri(self, uri):
         if uri_parts.query:
             self.extra = json.dumps(dict(parse_qsl(uri_parts.query, keep_blank_values=True)))
 
+    def get_uri(self) -> str:
+        uri = '{}://'.format(self.conn_type.replace('_', '-'))
+
+        authority_block = ''
+        if self.login is not None:
+            authority_block += quote(self.login, safe='')
+
+        if self.password is not None:
+            authority_block += ':' + quote(self.password)
+
+        if authority_block > '':
+            authority_block += '@'
+
+            uri += authority_block
+
+        host_block = ''
+        if self.host:
+            host_block += quote(self.host, safe='')
+
+            if self.port:
+                host_block += ':{}'.format(self.port)
+
+            if self.schema:
```

Comment:
Schema is like DB schema not scheme i.e. protocol
Just sticking with conn attr naming

Reply:
Oh you are saying we could have blank host but non blank schema...
This occured to me
We would need @ I think will try tomorrow

Reply:
Maybe best to go with simple alternative above, and weaken the tests so they don't care if empty login is parsed as '' or None?

Reply:
ok @ashb I have updated to support (with tests) for schema only, login only, password only, and port only

Reply:
ok lastly (one can hope) i have added another set of tests verifying that when we create connection from init attributes (rather than uri) that everything still works consistently -- URIs generated still parse to an equivalent connection object.